src/language_server_protocol.rs

```diff
@@ -201,8 +212,48 @@ impl LanguageClient {
             ),
         };
 
+        let mut semanticHighlightMaps = HashMap::new();
+
+        for (languageId, v) in userSemanticHighlightMaps {
+            let mut mapping_pairs = Vec::new();
+            match v {
+                Value::Object(mapping) => {
+                    mapping_pairs
+                        .extend(mapping.into_iter().map(|(hl_group, val)| (val, hl_group)));
+                }
+                Value::Array(values) => {
```

Comment:
Let's start with supporting one style first and begin from there.

With either two styles or one style, serde would be able to deserialize json string to the variable, without manually unwrapping and mapping. The added benefits is serde would be able to tell where and why deserialize failed if there is any error.